### PR TITLE
add tensorflow operation SegmentSum

### PR DESCRIPTION
**Pull request by qiuxin2012**
## What changes were proposed in this pull request?

add tensorflow operation SegmentSum

## How was this patch tested?
unit tests

### CODE REVIEW

File: spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/SegmentSum.scala [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.Table
+
+import scala.reflect.ClassTag
+
+/**
+ * Computes the sum along segments of a tensor.
+ */
+class SegmentSum[T: ClassTag]()(implicit ev: TensorNumeric[T])
```
Comment (yiheng) [inline spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/SegmentSum.scala]: Can input tensor has different numeric type of the parameter?

Comment (qiuxin2012) [reply spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/ops/SegmentSum.scala]: No, they are the same type.

**Comment by yiheng** [comment]
looks good to me. Please merge it after pass the pr validation test

**Comment by qiuxin2012** [comment]
jenkins passed

**Pull request merged**